### PR TITLE
Consensus metrics

### PR DESCRIPTION
**Pull request by Jason-Zhangxin-Chen**
Hi,

This is the new metrics for tendermint consensus engine, it was tested in the e2e test network.
There was no unit test case to cover those former metrics of the other components, I just implement the simple unit test cases for the new metrics introduced by this change to improve the code coverage.
The ticket is here: https://github.com/clearmatics/autonity/issues/264

BR,
Jason

### CODE REVIEW

File: consensus/tendermint/core/timeout.go [hunk 1]
```diff
@@ -70,10 +74,25 @@ func (t *timeout) stopTimer() error {
...
+	switch t.step {
+	case propose:
+		tendermintProposeTimer.UpdateSince(t.start)
+		return
+	case prevote:
+		tendermintPrevoteTimer.UpdateSince(t.start)
+		return
+	case precommit:
+		tendermintPrecommitTimer.UpdateSince(t.start)
+		return
```
Comment (JekaMas) [inline consensus/tendermint/core/timeout.go]: Is it possible to remove all the returns here?

Comment (Jason-Zhangxin-Chen) [reply consensus/tendermint/core/timeout.go]: Thanks, Eugene. If there is no return, does go continue to match the following cases? I am happy to replace it with a break.

Comment (Jason-Zhangxin-Chen) [reply consensus/tendermint/core/timeout.go]: Return was removed now.

File: consensus/tendermint/core/timeout.go [hunk 2]
```diff
@@ -85,16 +101,35 @@ func (t *timeout) reset(s Step) {
 	t.timer = nil
 	t.started = false
 	t.step = s
+	t.start = time.Now()
```
Comment (b00ris) [inline consensus/tendermint/core/timeout.go]: I think time.Time{} is better for reset

Comment (Jason-Zhangxin-Chen) [reply consensus/tendermint/core/timeout.go]: Thank you, Booris. It was fixed.

File: consensus/tendermint/core/core.go [hunk 3]
```diff
@@ -244,8 +244,19 @@ func (c *core) commit() {
 	}
 }
 
+// Metric collecton of round change and height change.
+func (c *core) measureHeightRoundMetrics(round *big.Int) {
+	if round == common.Big0 {
+		tendermintHeightChangeMeter.Mark(1)
```
[the remark this answers is not in the document]
Comment (Jason-Zhangxin-Chen) [reply consensus/tendermint/core/core.go]: Thank you, Booris. It was fixed.

File: consensus/tendermint/core/core.go [hunk 4]
```diff
@@ -251,8 +251,21 @@ func (c *core) commit() {
...
 }
 
+// Metric collecton of round change and height change.
+func (c *core) measureHeightRoundMetrics(round *big.Int) {
+	if round == common.Big0 {
+		// in case of height change, round changed too, so count it also.
+		tendermintRoundChangeMeter.Mark(1)
+		tendermintHeightChangeMeter.Mark(1)
+	} else {
+		tendermintRoundChangeMeter.Mark(1)
```
Comment (JekaMas) [inline consensus/tendermint/core/core.go]: tendermintRoundChangeMeter.Mark(1) is called in if and else both. Could we extract this call outside of if statement and remove else completely?

File: consensus/tendermint/core/core.go [hunk 5]
```diff
@@ -251,8 +251,21 @@ func (c *core) commit() {
 	}
 }
 
+// Metric collecton of round change and height change.
+func (c *core) measureHeightRoundMetrics(round *big.Int) {
+	if round == common.Big0 {
```
Comment (yazzaoui) [inline consensus/tendermint/core/core.go]: Equality operator on pointers `big.NewInt(0) != common.Big0`

Comment (Jason-Zhangxin-Chen) [reply consensus/tendermint/core/core.go]: Thank you, Youssef, it is fixed now.

**Comment by Jason-Zhangxin-Chen** [comment]
Fixed the unit test failure introduced by conflict merge too.

**Comment by Jason-Zhangxin-Chen** [comment]
Fix the lint warning too.

**Pull request merged**